spec/builders/latexmk-spec.js

```diff
@@ -92,6 +93,43 @@ describe('LatexmkBuilder', () => {
       })
     })
 
+    it('fails with code 12 and various errors, warnings and info messages are produced in log file', () => {
+      filePath = path.join(fixturesPath, 'error-warning.tex')
+
+      waitsForPromise(() => {
+        return builder.run(filePath).then(code => {
+          exitCode = code
+          parsedLog = builder.parseLogFile(filePath)
+        })
+      })
+
+      runs(() => {
+        const messages = [
+          { type: 'Error', text: 'There\'s no line here to end' },
+          { type: 'Error', text: 'Argument of \\@sect has an extra }' },
+          { type: 'Error', text: 'Paragraph ended before \\@sect was complete' },
+          { type: 'Error', text: 'Extra alignment tab has been changed to \\cr' },
+          { type: 'Warning', text: 'Reference `tab:snafu\' on page 1 undefined' },
+          { type: 'Error', text: 'Class foo: Significant class issue' },
+          { type: 'Warning', text: 'Class foo: Class issue' },
+          { type: 'Warning', text: 'Class foo: Nebulous class issue' },
+          { type: 'Info', text: 'Class foo: Insignificant class issue' },
+          { type: 'Error', text: 'Package bar: Significant package issue' },
+          { type: 'Warning', text: 'Package bar: Package issue' },
+          { type: 'Warning', text: 'Package bar: Nebulous package issue' },
+          { type: 'Info', text: 'Package bar: Insignificant package issue' },
+          { type: 'Warning', text: 'There were undefined references' }
+        ]
+
+        for (const message of messages) {
+          expect(_.some(parsedLog.messages,
```

Comment:
I used `_.some` because LaTeX font info messages will also show up and I am not sure if those  are entirely consistent across platforms. Since that that is not is not really what spec was trying to test I figured that looping through that messages that the file was known to generate and using `_.some` to find them was the easiest way to ignore the superfluous messages.

I have no problem thinking about it a little more.